mypy/subtypes.py

```diff
@@ -204,7 +203,8 @@ def visit_tuple_type(self, left: TupleType) -> bool:
                 if right.args:
                     iter_type = right.args[0]
                 else:
-                    iter_type = AnyType()
+                    # todo: is type of this Any correct?
+                    iter_type = AnyType(TypeOfAny.from_error)
```

Comment:
I think that _ideally_ this should be ``assert False``, since all these types have one type parameter and types with incorrect type argument numbers (e.g. zero) should be "patched" in the third pass of semantic analysis. There were some efforts recently to weed out all instances with incorrect type argument count (including test fixtures), but maybe some are still there.